profile-widget/diveeventitem.cpp

```diff
@@ -158,6 +173,10 @@ void DiveEventItem::setupToolTipString()
 		/* Do we have an explicit cylinder index?  Show it. */
 		if (internalEvent->gas.index >= 0)
 			name += tr(" (cyl. %1)").arg(internalEvent->gas.index + 1);
+		icd_data icd_data;
+		if (isobaric_counterdiffusion(lastgasmix, mix, &icd_data))
+			name += tr("\nICD ΔHe=%1% ΔN₂=%2%>+%3").arg(icd_data.dHe / 10).arg(icd_data.dN2 / 10).arg(icd_data.dHe / 50);
```

Comment:
This is imho not 100% correct yet. I see the following result for a gas change from (18/45) to EAN50:
![grafik](https://user-images.githubusercontent.com/25316050/35188412-5fa95a0c-fe35-11e7-808e-23cb97998a6c.png)

- Values all should have nicer format with leading plus sign and with a precision of 0.1 -> "+45.0%"
This unfortunately doesn't work with .arg and I didn't find out how this works in Qt. Any idea?
- Last value has wrong sign and is missing the "%"
- Yes, this is nitpicking but can we put "ICD:"?

Concerning translation: Maybe better only translate the "ICD"?

Reply:
Great work, @atdotde, I will run this through a bunch of my trimix dives and see how it functions.

Reply:
From what I can see there are only two issues. And I have a comment. 

1. Comment: The convention for the other items in the info box is to use a colon, e.g. 'PO2: 1.2bar'. Seeing we are using a proportional font, maybe it will be more space-efficient if one uses colons in this line as well, instead of '='. The line is getting quite long.
2. Issue 1: As @sfuchs79 indicated, the sign of the last figure needs attention.
3. Issue 2: See attached dive log. The change from 27/23 to EAN50 does not get an ICD info line. 
(I hope the markdown for a list works for the above paragraph)

This is, if I understand correctly, because isobaric_counterdiffusion() was not triggered. After all the dN2 was zero. I would like the icd information even if the change was within the rule_of_fifths.

The second reason may be because an ascent was not triggered at this gas change. Frequently it is necessary to stop at a gas change, to perform a coordinated gas change with buddy or to pick up a deco cylinder in a cave. So there is a short stop during the gas change. We need to ensure that, in such cases, the ascent still applies. 

[example85.xml.tar.gz](https://github.com/Subsurface-divelog/subsurface/files/1649554/example85.xml.tar.gz)

Thank you again fir this work.